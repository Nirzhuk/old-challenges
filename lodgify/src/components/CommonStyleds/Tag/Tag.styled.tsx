

import styled from "styled-components";

export const backgroundColorsTags = {
    BOOKED: '#2C64B7',
    UNAVAILABLE: '#E02323',
    AVAILABLE: '#11A960'
};

export const TagStyled = styled.span<any>`
    padding: 3px 10px;
    margin: 8px 0px 0px 8px;
    border-radius: 30px;
    line-height: 20px;
    font-weight: 700;
    font-size:12px;
    width: fit-content;
    line-height:15px;
    border-top: 0;
    color:#fff;
    border-right: 0;
    background-color: ${({ stateColor }: any) => stateColor};
  `;
