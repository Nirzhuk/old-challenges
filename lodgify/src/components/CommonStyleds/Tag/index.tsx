import React from 'react'
import { BookableStatusType } from '../../../core/types/App'
import { backgroundColorsTags, TagStyled } from './Tag.styled'

interface TagProps {
    children: React.ReactNode;
    state: BookableStatusType;
}
const Tag = ({ children, state }: TagProps) => {

    const getStateColor = (state: string) => {
        switch (state) {
            case 'BOOKED':
                return backgroundColorsTags[state]
            case 'UNAVAILABLE':
                return backgroundColorsTags[state]
            case 'AVAILABLE':
                return backgroundColorsTags[state]

            default:
                return backgroundColorsTags['BOOKED']
        }
    }
    return (
        <TagStyled stateColor={getStateColor(state)} data-testid="tag">
            {children}
        </TagStyled>
    )
}

export default Tag
