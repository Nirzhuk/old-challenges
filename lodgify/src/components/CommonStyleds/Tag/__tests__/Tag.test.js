import { render, screen } from '@testing-library/react'
import Tag from '..';


test('Booked tag', async () => {
    const { container} = render(<Tag state="BOOKED">booked</Tag>)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByTestId("tag")).toHaveTextContent(`booked`)
    expect(screen.getByTestId("tag")).toHaveStyle(`background-color: #2C64B7`)

})
test('available tag', async () => {
    const { container } = render(<Tag state="AVAILABLE">unavailable</Tag>)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByTestId("tag")).toHaveTextContent(`unavailable`)
    expect(screen.getByTestId("tag")).toHaveStyle(`background-color: #11A960`)
})

test('unavailable tag', async () => {
    const { container } = render(<Tag state="UNAVAILABLE">available</Tag>)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByTestId("tag")).toHaveTextContent(`available`)
    expect(screen.getByTestId("tag")).toHaveStyle(`background-color: #E02323`)
})