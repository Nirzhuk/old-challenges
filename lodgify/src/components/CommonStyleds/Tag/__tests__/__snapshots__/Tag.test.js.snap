// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booked tag 1`] = `
<span
  class="sc-bdnxRM iafYIX"
  data-testid="tag"
>
  booked
</span>
`;

exports[`available tag 1`] = `
<span
  class="sc-bdnxRM iOncZf"
  data-testid="tag"
>
  unavailable
</span>
`;

exports[`unavailable tag 1`] = `
<span
  class="sc-bdnxRM eDRCdU"
  data-testid="tag"
>
  available
</span>
`;
